### PR TITLE
add vector support to assume and assumeAlso

### PR DESCRIPTION
**Pull request by cbm755**
This is a rebase and bug fix of #737.

### CODE REVIEW

**Comment by cbm755** [comment]
I also rewrote the commits to make it clearer this is a minimal change.  I did this by doing the pure indent changes as separate commits.

**Comment by cbm755** [comment]
@latot  ok to merge?

File: inst/@sym/assume.m [hunk 1]
```diff
@@ -116,47 +116,48 @@
...
     end
-    newx = sym(xstr, ca);
-  end
 
-  if (nargout > 0)
-    varargout{1} = newx;
-    return
-  end
+    if (nargout > 0)
+      varargout{i} = newx;
```
Comment (latot) [inline inst/@sym/assume.m]: little question here, if we do
```
syms x
f=x*2;
x = assume(x, 'positive')
```
```f``` will be updated with the new assumptions?

if not will be updated i think is missed the ```return```, and if not is fine.

(This for assume and assumeAlso)

Thx. Cya.

Comment (cbm755) [inline inst/@sym/assume.m]: `f` should not be updated.  `assumptions(f)` should be empty.  I believe this is tested.

I don't think the return is missing: the "mucking around" bit is in a "else" block.

Comment (cbm755) [inline inst/@sym/assume.m]: Thanks for looking at this.  I've double-checked and those cases are indeed tested.

**Comment by latot** [comment]
Hi, the last thing, can you add one more test to check assume and assumeAlso with 2 new assuptions please?

```
assume(x, 'even', 'positive')
assumeAlso(x, 'even', 'positive')
```

like that, i remember there was some weird things, only to test it.

Thx. Cya.

**Comment by cbm755** [comment]
Ok, done.

(The "weird things" are probably `'real', 'positive'` and `'even', 'real'` where `'real'` is redundant and sometimes non-deterministically disappears #739.)

**Pull request merged**